lib/cve.py

```diff
@@ -51,82 +50,127 @@ def read_json(file):
 def get_component_names(data_list):
     component_list = []
     for component_info in data_list:
+        log(f"component_info: {component_info}")
         for key, value in component_info.items():
             if (key == "name"):
                 component_list.append(value)
     return component_list
 
 
 def get_component_detail(data_list, component):
-    found = 0
+    log(f"looking for component detail: {component}")
     for component_info in data_list:
-        for key, value in component_info.items():
-            if (key == "name") and (value == component):
-                found = 1
-            if (key == "source"):
-                return(value["git"]["url"],value["git"]["revision"])
+        log(f"component_info: {components_info}")
+        if component == component_info["name"]:
+            return component_info["source"]["git"]["url"], component_info["source"]["git"]["revision"]
```

Comment:
Below you created methods for 'get_snapshot_name' and 'get_snapshot_namespace'. It seems like you should create a couple of additional methods so you can use 'get_component_url(component_info)' and 'get_component_revision(component_info)' here. That would also allow you to cleanly check for the existence of the field in the structures and provide a clear error message if they don't exist.

Reply:
fixed